src/components/Upload/index.jsx

```diff
@@ -110,15 +110,18 @@ class Upload extends React.Component {
         // Once upload is done, create a resource
         const ckanResource = frictionlessCkanMapper.resourceFrictionlessToCkan(
           resource.descriptor
-        );
-        delete ckanResource.sample;
-        client.action(
-          "resource_create",
-          Object.assign(ckanResource, {
-            package_id: this.state.datasetId,
+          );
+          delete ckanResource.sample;
+          client.action(
+            "resource_create",
+            Object.assign(ckanResource, {
```

Comment:
This kind of stuff should happen *outside* of a method call. Do the assignment before the call, and it would also be cleaner to not modify the underlying object but create a copy, add stuff and then save.

Reply:
Yes, totally agree.